guide/src/android/foreground-service.md

```diff
@@ -0,0 +1,116 @@
+# Android Foreground Service
+
+An android foreground service is provided that runs automatically during a navigation trip. This complies with Google's guidance on continued location access [Foreground Location Requirements](https://support.google.com/googleplay/android-developer/answer/9799150#Accessing%20location%20in%20the%20foreground).
+
+## Overview
+
+The Kotlin `ForegroundServiceManager` links `FerrostarCore` to the foreground service `FerrostarForegroundService` using a binder. This technique allows us to bind the foreground service to a navigation trip operated by the core. When navigation starts, the foreground service is started and bound. When navigation is stopped, we stop the foreground and unbind it. Note, you must call stop on ferrostar core to stop the notification. It does not automatically close out when the user arrives because location updates are not stopped.
+
+### Setting Up Permissions
+
+This feature uses the existing `AndroidManifest.xml` items in the ferrostar core module. These do not need to be added to your app.
+
+```xml
+    <!-- Foreground service permission -->
+    <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_LOCATION" />
+
+    <application>
+        <service
+            android:name="com.stadiamaps.ferrostar.core.service.FerrostarForegroundService"
+            android:foregroundServiceType="location"
+            />
+    </application>
+```
+
+Your app must still request `POST_NOTIFICATIONS` and `FOREGROUND_SERVICE_LOCATION` permissions on API 34+ (Upside Down Cake). See the demo app's call to request these permissions in your composable app.
+
+```kotlin
+  val allPermissions =
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+        arrayOf(
+            Manifest.permission.ACCESS_FINE_LOCATION,
+            Manifest.permission.ACCESS_COARSE_LOCATION,
+            Manifest.permission.POST_NOTIFICATIONS,
+            Manifest.permission.FOREGROUND_SERVICE_LOCATION)
+      } else {
+        arrayOf(
+            Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION)
+      }
+
+  val permissionsLauncher =
+      rememberLauncherForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) {
+        // TODO: Handle permission fialures.
+      }
+```
+
+### Adding the Manager to the Core.
+
+In your app module, you'll need to pass in the `ForegroundServiceManager` as a parameter to construct `FerrostarCore`. Here's how to do that with the included notification builder.
```

Comment:
Note: will need to loosen this language if the foreground service doesn't end up being a hard requirement.

```suggestion
In your app module, you'll need to pass in the `ForegroundServiceManager` as a parameter to construct `FerrostarCore`.
Here's how to do that with the included notification builder.
```